api/envoy/config/core/v3/base.proto

```diff
@@ -245,15 +245,17 @@ message Metadata {
   // :ref:`typed_filter_metadata <envoy_v3_api_field_config.core.v3.Metadata.typed_filter_metadata>`
   // fields are present in the metadata with same keys,
   // only ``typed_filter_metadata`` field will be parsed.
-  map<string, google.protobuf.Struct> filter_metadata = 1;
+  map<string, google.protobuf.Struct> filter_metadata = 1
+      [(validate.rules).map = {keys {string {min_len: 1}}}];
 
   // Key is the reverse DNS filter name, e.g. com.acme.widget. The ``envoy.*``
   // namespace is reserved for Envoy's built-in filters.
   // The value is encoded as google.protobuf.Any.
   // If both :ref:`filter_metadata <envoy_v3_api_field_config.core.v3.Metadata.filter_metadata>`
   // and ``typed_filter_metadata`` fields are present in the metadata with same keys,
   // only ``typed_filter_metadata`` field will be parsed.
-  map<string, google.protobuf.Any> typed_filter_metadata = 2;
+  map<string, google.protobuf.Any> typed_filter_metadata = 2
+      [(validate.rules).map = {keys {string {min_len: 1}}}];
```

Comment:
I am thinking `""` is also a valid key for metadata like `mymap[""]="foo"`, and this probably will break some use cases. Could you also link the fuzzer issue? cc @adisuissa.

Reply:
+1 on adding fuzz test case.

@botengyao I'm not sure if an empty string is a valid case. Have you seen/used a config with an empty key?

Reply:
At least the e.g., `dynamicMetaData` in stream info is using this protocol, it is safe to set to empty string, does PGV impact it?

Reply:
Sorry, I don't follow.
In a filter_metadata/typed_filter_metadata the keys need to be the reverse DNS filter name, no?
Can you point me to the example where the keys are empty?

Reply:
I agree that it is a core type that is being used in many places.
AFAIK the idea is to later search the metadata for the relevant key (e.g., [here](https://github.com/envoyproxy/envoy/blob/17b1db5bc290a5be36b5065848cec07c26e764f4/source/common/router/config_impl.cc#L555-L556)), and if that key is empty, then it seems that the search will not work properly.
If you know of any other uses of this field, please let us know, so we can fix them.

Reply:
sgtm, and the search can also be customized to match the `""` by some extensions,  but agreed that we should fix it if that happens, thanks!